### PR TITLE
Add neuron publishing & refactor for consistent directory/file names

### PR DESCRIPTION
**Pull request by srid**
- Add neuron configuration
- Rename folders to match filename
- Add github workflow to publish

**Preview** (wait a few mins): https://srid.github.io/obsidian-haskell/impulse

This PR renames a few folders to match the Markdown file (eg: "Company" -> "Companies" to match "Companies.md"). Why? Because:

- In Obsidian published view, you can click on the folder in a sidebar to view the "parent" note. Instead of the parent note appearing dangling separate from the folder note.
- In neuron's [dirtree](https://neuron.zettel.page/dirtree) plugin this gives us a free hierarchy without having to manually link to the children.

### CODE REVIEW

**Comment by srid** [comment]
Ready: https://srid.github.io/obsidian-haskell/impulse

(If there is an `index.md` that would be show up when accessing the root URL)

File: .github/workflows/neuron.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: "Publish Neuron site"
+on:
+  # Run only when pushing to master branch
+  # TODO: Remove "neuron" after merge
+  push:
+    branches:
+      - master
+      - neuron
```
Comment (tfausak) [inline .github/workflows/neuron.yaml]:
```suggestion
  push:
    branches:
      - main
```

**Comment by srid** [comment]
btw, there is no obligation to merge this :-) just wanted to throw something for demo and invite comments from anyone if any.

the directory/file convention however would be a good thing to ensure in regards to being "a data source that others are encouraged to publish" https://github.com/tfausak/obsidian-haskell/issues/2#issuecomment-824088430

**Review by tfausak — COMMENTED** [review]
Thanks for opening this PR! 

I didn't know about the directory naming scheme (778919227fba3f5e765801407478d3f2ea380ea7). That's a good change. 

I like how the generated site is static, which would help with #2. That being said, it's not quite as featureful as the Obsidian site. For the time being I personally only want to support one deployment, so I think I'll stick with Obsidian. I have two thoughts about moving forward though:

- I'd be happy to posture this repo as a data source that others are explicitly encouraged to render and host however they'd like. 
- If this gets traction with the broader community, it'll probably be worth the effort to maintain many deployments with various feature sets. (Or develop one deployment that checks all the boxes.)

**Comment by tfausak** [comment]
After renaming this project (163b3179aa2a4191f78e009f37926d2908f7fb09), this is now available at: https://taylor.fausak.me/haskell-knowledge-base/

**Pull request merged**